### DISCUSSION TITLE
Loading JS (when using navigate in URI)

### DISCUSSION THREAD

**Discussion by Tresor-Kasenda** [livewire/livewire#7956] [1. Help]
Hi, I am facing an issue when I am using wire:navigate on my links the js is not loading, reading the documentation we suggest to put the data-navigate-track attribute on the script tag <script src="{{ asset('scripts.js') }}" data-navigate-track></script> it still does not work, how can I fix this issue?
![code](https://github.com/livewire/livewire/assets/34010260/9af22571-bf39-47fb-80ea-ce14467aebae)

**Comment by danie-ramdhani** [comment]
no, `data-navigate-track` is to track the script hash when it's change (https://livewire.laravel.com/docs/navigate#reloading-when-assets-change) i.e. this `app.js` if the `id` is change in page two, it will request the new `app.js` script.

```blade
<!-- Page one -->
<head>
    <script src="/app.js?id=123" data-navigate-track></script>
</head>
 
<!-- Page two -->
<head>
    <script src="/app.js?id=456" data-navigate-track></script>
</head>
```

maybe you're looking with livewire browser event (https://livewire.laravel.com/docs/javascript#global-livewire-events) like `livewire:init` or `livewire:navigated`?

```js
    document.addEventListener('livewire:init', () => {
        // Runs after Livewire is loaded but before it's initialized
    })
    
    // another one is navigated, if the script needs to run after navigated to another page
   document.addEventListener('livewire:navigated', () => {
      // 
   })
```

**Reply by Tresor-Kasenda** [reply]
are you an idea about how to load `<script src="script.js"></script>` inside of ```document.addEventListener('livewire:navigated', () => {})```
If yes you can help me

**Reply by codeilo** [reply]
I am also having the same issue.
